test/ng/directive/ngEventDirsSpec.js

```diff
@@ -150,4 +150,38 @@ describe('event directives', function() {
     }));
 
   });
+
+  describe('click', function() {
+
+    it('should call the listener synchronously if inside of $apply',
+        inject(function($rootScope, $compile) {
+      element = $compile('<button type="button" ng-click="click()">Button</button>')($rootScope);
+      $rootScope.click = jasmine.createSpy('click');
+
+      $rootScope.$apply(function() {
+        element.triggerHandler('click');
+      });
+
+      expect($rootScope.click).toHaveBeenCalledOnce();
```

Comment:
This doesn't actually check that it is called synchronously (e.g. it might be called outside of the `$apply()` block).
You could use the same approach as below (using a `$watch`).